tests/parser/functions/test_convert_to_bytes32.py

```diff
@@ -80,3 +80,107 @@ def testConvertBytes32(flag: bool) -> bytes32:
     trueBytes = c.testConvertBytes32(True)
     assert trueBytes[31:32] == b'\x01'
     assert len(trueBytes) == 32
+
+
+#################################################################################
+# NOTE: Vyper uses a decimal divisor of 10000000000 (or 10^10).
+#
+#       This means that `decimal` type variables can store values
+#       that are of 1/10000000000.
+#
+#       Because of this, when converting from `decimal` to `bytes32`,
+#       the conversion can be thought of as converting integer result of
+#       the decimal value of interest multiplied by 10000000000.
+#
+#       For example, converting the decimal value `5.0` to `byte32`
+#       can be thought of as giving the `bytes32` value of the integer
+#       result of 5 * 10000000000 = 50000000000
+#################################################################################
+def test_convert_from_decimal(get_contract_with_gas_estimation):
+    code = """
+bar: decimal
+nar: decimal
+mar: decimal
+jar: decimal
+kar: decimal
+
+@public
+def foo() -> bytes32:
+    return convert(0.0, bytes32)
+
+@public
+def foobar() -> bytes32:
+    self.bar = 0.0
+    return convert(self.bar, bytes32)
+
+@public
+def hoo() -> bytes32:
+    return convert(MIN_DECIMAL, bytes32)
+
+@public
+def hoonar() -> bytes32:
+    self.nar = MIN_DECIMAL
+    return convert(self.nar, bytes32)
+
+@public
+def goo() -> bytes32:
+    return convert(MAX_DECIMAL, bytes32)
+
+@public
+def goomar() -> bytes32:
+    self.mar = MAX_DECIMAL
+    return convert(self.mar, bytes32)
+
+@public
+def zoo() -> bytes32:
+    return convert(5.0, bytes32)
+
+@public
+def zoojar() -> bytes32:
+    self.jar = 5.0
+    return convert(self.jar, bytes32)
+
+@public
+def xoo() -> bytes32:
+    return convert(-5.0, bytes32)
+
+@public
+def xookar() -> bytes32:
+    self.kar = -5.0
+    return convert(self.kar, bytes32)
+    """
+
+    c = get_contract_with_gas_estimation(code)
+    decimal_divisor = 10000000000
+
+    fooVal = c.foo()
+    foobarVal = c.foobar()
+    assert fooVal == (b"\x00" * 32)
+    assert len(fooVal) == 32
+    assert foobarVal == (b"\x00" * 32)
+    assert len(foobarVal) == 32
+    assert fooVal == foobarVal
+
+    hooVal = c.hoo()
+    hoonarVal = c.hoonar()
+    _hoo = ((-2**127) * decimal_divisor).to_bytes(32, byteorder="big", signed=True)
```

Comment:
Maybe we should factor out these repeated `.to_bytes(...)` operations into a helper function at the top of this testing module.  Also, you can import `DECIMAL_DIVISOR` from `vyper.utils`.